source/common/network/cidr_range.h

```diff
@@ -117,6 +120,23 @@ class CidrRange {
   int length_;
 };
 
+/**
+ * Class for keeping a list of CidrRanges, and then determining whether an
+ * IP address is in the CidrRange list.
+ */
+class IpList {
+public:
+  IpList(const std::vector<std::string>& subnets);
+  IpList(const Json::Object& config, const std::string& member_name);
```

Comment:
Should this just be a static factory? It seems that logically the only constructor we should have is the one that takes the subnets.